### ISSUE TITLE
Remove SlimServeAsset's dependency on the assets service

### ISSUE THREAD

**Issue by alexweissman** [userfrosting/assets#4]
`SlimServeAsset` seems to depend on the `assets` service, defined in the core Sprinkle's `ServiceProvider`.  This is a problem for two reasons:

1. You have code in a dependency (this package) that is, in fact, dependent on code in the very project that is relying on this package as a dependency. 😵 
2. The `assets` service gets extended in the `account` Sprinkle to load additional asset bundles for user themes. This requires a database query. By making the serving of individual assets (that was formerly handled by `AssetLoader`) dependent on the new, unified assets service, you're causing the database to get queried for each and every asset load. The original idea of having separate services, was that you only needed to query the database to determine _which_ assets to load (and generate the necessary HTML tags), not to actually _perform_ the loads themselves.

This is exactly the problem I ran into over a year ago, and prompted the rewrites that I did in:
https://github.com/userfrosting/assets/commit/f693c9f98af361bdb90259fac0813753356ee03e#diff-f92daec31f2307ec233bd3bbc6eb16d0
and 
https://github.com/userfrosting/UserFrosting/commit/02ec7731c9fbe0a839be04ea25d3e60f6238ff86

**Comment by lcharette** [comment]
`SlimServeAsset` is not actually used in `assets`. Actually, I'm not even sure `SlimServeAsset` is used anywhere :notsureif:

**Comment by Silic0nS0ldier** [comment]
https://github.com/userfrosting/UserFrosting/commit/f4e43c7df7520025cd0ff1ef36759ee65027abc2

**Comment by Silic0nS0ldier** [comment]
Bingo: https://github.com/userfrosting/assets/blob/develop/src/ServeAsset/SlimServeAsset.php#L53

**Comment by lcharette** [comment]
Before I forget 

> So the goal was to get rid of `AssetLoader`, and doing that `SlimServeAsset` was introduced... That #4 issue is that `SlimServeAsset` *should* be in core, so removed from `assets`. And meanwhile, in a *separate* issue, `AssetLoader` should be actually deprecated by somehow just being loaded into that `getAsset` route method ?

https://chat.userfrosting.com/group/dev?msg=dMRG9jL6ouGcdjhaq

> the reason to have it as a separate service and class, was to avoid excessive dependencies that end up slowing down each request for assets. (or that require accessing the current client session)

So plan to close this issue is : 
1. Move `AssetLoader` back into `assets`. It doesn't really needs to be deprecated, as `getContent()` is useful. Make it depend on [`Asset`](https://github.com/userfrosting/assets/blob/develop/src/Assets.php#L269), which is also in `assets`, instead of the whole `ci`. 
1. Optionally delete `SlimServeAsset`. Routes method belong in `core`, but we _could_ leave it here if it now depends on the `AssetLoader` instead of `ci`.
1. Optimize `getAsset` route so it doesn't load unwanted dependencies and touch the user for every request, resulting in session bullying (Actually open new issue about this if it's still an issue and close this one)


tl;dr : `SlimServeAssets` should depends on `AssetLoader` and `AssetLoader ` should depend on `Assets`, not `ci`.

**Comment by lcharette** [comment]
Should be fixed in `feature-locator` branch. Will merge into `develop` soon
